flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java

```diff
@@ -304,6 +304,35 @@ void testTransformationSetMaxParallelism() {
                 .isEqualTo(20);
     }
 
+    /**
+     * Tests that the maxParallelism for an operator chain is correctly determined as the minimal
+     * maxParallelism of all operators in the chain.
+     */
+    @Test
+    void testTransformationSetMinimalMaxParallelismForChain() {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setMaxParallelism(16);
+
+        DataStreamSource<Long> source = env.fromSequence(1L, 3L);
+        source.setParallelism(1)
+                .setMaxParallelism(128)
+                .map(i -> i)
+                .setParallelism(1)
+                .setMaxParallelism(1)
+                .print()
+                .setParallelism(1);
```

Comment:
Would it also make sense to test the reverse order: 
source.maxParallelism(1)
sink.maxParallelism(128)
?